ddtrace/tracer/option.go

```diff
@@ -186,16 +187,23 @@ func WithPropagator(p Propagator) StartOption {
 	}
 }
 
-// WithServiceName sets the default service name to be used with the tracer. It is
-// deprecated in favour of WithService and will be removed in the next major version.
+// WithServiceName sets the default service name to be used with the tracer. This will
+// not affect the service name used by any integrations. It is deprecated in favour of
+// WithService and will be removed in the next major version.
 func WithServiceName(name string) StartOption {
-	return WithService(name)
+	return func(c *config) {
+		c.serviceName = name
+		globalconfig.SetServiceName("")
```

Comment:
This seems hacky. Why do we need it? Let's drop it. We don't care if the user uses both of these options. The documentation should clearly explain the behaviour.

Reply:
I think we said we would print a warning instead? Although I'm not sure if that will be visible enough to register.

Reply:
I'm on the fence. I still think we should remove it. `DD_SERVICE` is equivalent to using the `WithService` option. The counterpart for `WithServiceName` would be `DD_SERVICE_NAME` (which we don't have). I doubt someone will have `DD_SERVICE` different from `DD_SERVICE_NAME`.

Reply:
FWIW A warning sounds good too!

Reply:
If I move the `globalconfig.SetServiceName(...)` out of here and into `tracer.Start`, I won't be able to detect this situation to either unset it *or* warn about it.

The best we can do is add a warning any time `WithServiceName` is called about it being deprecated.